dsc/credentialEncryption.md

```diff
@@ -0,0 +1,317 @@
+ 4. Run the DSC configuration for the _target node_ on the authoring computer and apply it to the _target node_.
+ 5. Create a Local Configuration Manager (LCM) configuration script to configure the _target node_ LCM to use the encryption certificate (as identified by its thumbprint) to decrypt DSC configuration credentials, then apply the LCM configuration to the _target node_ and start DSC configuration.
+ 
+![Diagram1](images/CredentialEncryptionDiagram1.png)
+
+## Certificate Requirements
+To enact credential encryption, a public key certificate is required to be available on the _Target Node_ that is **trusted** by the computer being used to author the DSC configuration.
+This public key certificate has specific requirements that must be satisfied to allow it to be used for DSC Credential Encryption:
+ 1. **Key Usage**: must contain 'DigitalSignature', 'KeyEncipherment' and 'DataEncipherment'.
+ 2. **Enhanced Key Usage**: must contain 'Document Encryption'.
+ 3. The Private Key for the certificate is available on the _Target Node_.
+
+## Configuration Data
+When dealing with encryption of DSC configuration, it is important to understand the DSC **configuration data** structure.
+This structure is used to separate configuration data from the logic of a configuration file.
+It allows node specific configuration data to be passed to a DSC configuration when it is run.
+See [Separating Configuration and Environment Data](configData.md) for more information on the **configuration data** structure. 
+
+The elements that can be configured for each node that are related to credential encryption are:
+* **NodeName** - the name of the target node that the credential encryption is being configured for.
+* **PsDscAllowPlainTextPassword** - whether unencrypted credentials will be allowed to be passed to this node. This is **not recommended**.
+* **Thumbprint** - the thumbprint of the certificate that will be used to decrypt the credentials in the DSC Configuration on the _Target Node_. **This certificate must exist in the Local Machine certificate store on the Target Node.**
+* **CertificateFile** - the certificate file (containing the public key only) that should be used to encrypt the credentials for the _Target Node_.
+
+The following is an example of **configuration data** for a single node named `targetNode` which should be encrypted using a public key certificate that has been exported to `C:\publicKeys\targetNode.cer`:
+
+```PowerShell
+$ConfigData = @{ 
+        AllNodes = @(     
+                        @{  
+                            # The name of the node we are describing 
+                            NodeName        = “targetNode”
+
+                            # The path to the .cer file containing the 
+                            # public key of the Encryption Certificate 
+                            # used to encrypt credentials for this node 
+                            CertificateFile = “C:\publicKeys\targetNode.cer”
```

Comment:
There were questions on the blog if you could also use a local cert in the store.  the answer is yes, but you should avoid transporting the private key protected with a well known password.  In most cases where i asked how they would implement this, the answer came down to this implementation.